pkg/kn/commands/source/cronjob/cronjob.go

```diff
@@ -0,0 +1,46 @@
+// Copyright © 2019 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cronjob
+
+import (
+	"github.com/spf13/cobra"
+
+	"knative.dev/client/pkg/kn/commands"
+)
+
+func NewCronJobCommand(p *commands.KnParams) *cobra.Command {
+	cronImporterCmd := &cobra.Command{
+		Use:   "cronjob",
+		Short: "CronJob source command group",
+	}
+	cronImporterCmd.AddCommand(NewCronJobCreateCommand(p))
+	cronImporterCmd.AddCommand(NewCronJobDeleteCommand(p))
+	cronImporterCmd.AddCommand(NewCronJobDescribeCommand(p))
+	cronImporterCmd.AddCommand(NewCronJobUpdateCommand(p))
+	return cronImporterCmd
+}
+
+func newCronJobSourceClient(p *commands.KnParams, cmd *cobra.Command) (*CronSourceClient, error) {
+	namespace, err := p.GetNamespace(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	config, err := p.GetClientConfig()
```

Comment:
adding a note here for related change in another PR, have added `prepareRestConfig` [here](https://github.com/knative/client/pull/536/files#diff-52c760be60a1c01977e8e9bcac3167c3R108) , avoids further processing of client config and can directly be passed for creating clientset, we can refine later for aligning this.

Reply:
+1 I updated to move `prepareRestConfig` to `RestConfig` so that I can access it from here. The issue why I'm not using `param.NewSourcesClient()` and `KnTestCommand` here is, that ...

* ... it does not really deal well when two fake clients are needed. That ApiServerSource create tests work is by accident, as the serving FakeService is not prepared and just returns an empty service when the create command looks up the sink.
* ... the approach doesn't work well with the mock tests (there are preparing the commands on their own to inject the Mockclient instead of a fake service)
* ... its IMO better decoupled if every command can look up its client on its own (which makes later refactoring easier and one does not always have to touch top-level components like KnParam to introduce new features)


I suggest to merge this PR nevertheless now, and I will immediately jump on to refactor the multi client management afterwards.

Reply:
Alright! :+1: